scripts/generate_source.py

```diff
@@ -0,0 +1,150 @@
+    args = parser.parse_args(argv)
+
+    gen_spec_cmds = [*[[repo_relative('scripts/vt_genvk.py'),
+                   '-registry', os.path.abspath(os.path.join(args.registry, 'vk.xml')),
+                   '-scripts', os.path.abspath(args.registry),
+                   '-quiet',
+                   filename] for filename in ["api_dump.cpp",
+                                              "api_dump_text.h",
+                                              "api_dump_html.h",
+                                              "api_dump_json.h"]]]
+
+    gen_video_cmds = [*[[repo_relative('scripts/vt_genvk.py'),
+                   '-registry', os.path.abspath(os.path.join(args.registry, 'video.xml')),
+                   '-scripts', os.path.abspath(args.registry),
+                   '-quiet',
+                   filename] for filename in ["api_dump_video_text.h",
+                                              "api_dump_video_html.h",
+                                              "api_dump_video_json.h"]]]
+
+    gen_cmds = gen_spec_cmds + gen_video_cmds
+
+    repo_dir = repo_relative('layersvt/generated')
+
+    # Update the api_version in the respective json files
+    if args.generated_version:
+        json_files = []
+        json_files.append(repo_relative('layersvt/json/VkLayer_api_dump.json.in'))
+        for json_file in json_files:
+            with open(json_file) as f:
+                data = json.load(f)
+
+            data["layer"]["api_version"] = args.generated_version
+
+            with open(json_file, mode='w', encoding='utf-8', newline='\n') as f:
+                f.write(json.dumps(data, indent=4))
+
+    # get directory where generators will run
+    if args.verify or args.incremental:
+        # generate in temp directory so we can compare or copy later
+        temp_obj = tempfile.TemporaryDirectory(prefix='vt_codegen_')
+        temp_dir = temp_obj.name
+        gen_dir = temp_dir
+    else:
+        # generate directly in the repo
+        gen_dir = repo_dir
+
+    # run each code generator
+    for cmd in gen_cmds:
+        print(' '.join(cmd))
+        try:
+            subprocess.check_call([sys.executable] + cmd, cwd=gen_dir)
+        except Exception as e:
+            print('ERROR:', str(e))
+            return 1
+
+    # optional post-generation steps
+    if args.verify:
+        # compare contents of temp dir and repo
+        temp_files = set(os.listdir(temp_dir))
+        repo_files = set(os.listdir(repo_dir))
+        files_match = True
+        for filename in sorted((temp_files | repo_files) - set(verify_exclude)):
+            temp_filename = os.path.join(temp_dir, filename)
+            repo_filename = os.path.join(repo_dir, filename)
+            if filename.startswith('gpu_'):
+                # The shaders requires glslangvalidator,
+                # so updated manually with generate_spirv when needed
+                continue
```

Comment:
This looks like a special case that VVL added? I don't think it does anything in this repo

Reply:
Ah!  Good catch.  I also found a few lines I'd nuked but not deleted.  Cleaned up a bit.  One other weird thing -- using `video.xml` with the V-H registry generates these scary warning messages for every file: `WARNING: No matching API versions found!`.  Is it worth a note in the readme or something?  It all works fine, just spits out spam.

Reply:
I've been ignoring the `WARNING: No matching API versions found!` because it seems to work just fine. The generator script for api dump works whether the API versions are there or not after all.

Reply:
The Vulkan headers build throws the exact same error when building the video headers
```
$ make -C xml codec_headers
make: Entering directory '/home/mike/Projects/vulkan/xml'
WARNING: No matching API versions found!
WARNING: No matching API versions found!
WARNING: No matching API versions found!
WARNING: No matching API versions found!
WARNING: No matching API versions found!
WARNING: No matching API versions found!
WARNING: No matching API versions found!
WARNING: No matching API versions found!
WARNING: No matching API versions found!
make: Leaving directory '/home/mike/Projects/vulkan/xml'
```
Just looks like a wart from the way that they reused the reg.py tooling for video headers which only really contain typedefs.